File: smart-campus-backend/src/main/java/com/smartcampus/notification/service/NotificationService.java
package com.smartcampus.notification.service;

import com.smartcampus.notification.dto.NotificationResponse;
import com.smartcampus.notification.enums.NotificationType;

import java.util.List;

public interface NotificationService {
    void create(Long recipientUserId, NotificationType type, String message, Long referenceId);
    List<NotificationResponse> getForUser(Long userId);
    void markRead(Long notificationId);
    void markAllRead(Long userId);
    long getUnreadCount(Long userId);
}

File: smart-campus-frontend/src/services/notification.service.ts
import axios from 'axios';
import { getApiBaseURL } from '../config/apiBase';
import type { NotificationResponse } from '../types/notification.types';

const api = axios.create({
  baseURL: getApiBaseURL(),
  headers: { 'Content-Type': 'application/json' },
});

api.interceptors.request.use(config => {
  const token = localStorage.getItem('sc_access_token');
  if (token) config.headers.Authorization = `Bearer ${token}`;
  return config;
});

const BASE = '/api/notifications';

export const notificationService = {
  getForUser(userId: number): Promise<NotificationResponse[]> {
    return api.get<NotificationResponse[]>(BASE, { params: { userId } }).then(r => r.data);
  },

  getUnreadCount(userId: number): Promise<number> {
    return api.get<number>(`${BASE}/unread-count`, { params: { userId } }).then(r => r.data);
  },

  markRead(id: number): Promise<void> {
    return api.patch(`${BASE}/${id}/read`).then(() => undefined);
  },

  markAllRead(userId: number): Promise<void> {
    return api.patch(`${BASE}/mark-all-read`, null, { params: { userId } }).then(() => undefined);
  },
};
